google/cloud/apigee/registry/applications/v1/diff_analytics.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+package google.cloud.apigee.registry.applications.v1;
+// [START go_declaration]
+option go_package = "github.com/apigee/registry/rpc;rpc";
+// [END go_declaration]
+
+/* What the Differ Module will compute, this proto contains
+the text diff of a spec and its revision */
+message TextDiff {
+  string diff = 1;
+  string spec = 2;
+  string spec_revision = 3;
+}
+
+/* What the Breaking Change Module will compute, this proto contains
+the diff separated into lines which represent individual changes.
+These changes are then labeled as a breaking change or non breaking change. */
+message DiffAnalysis {
+  //A single change
+  message ChangeLine{
+    enum ChangeType {
+      ADDED = 0;
+      MODIFIED = 1;
+      DELETED = 2;
+    }
+    /* A string line that represents 1 change
+   Ex : Components: schemas: modified: Pet: properties:
+   modified: name: type: from: string: to: "null" */
+    string line = 1;
+    //Added, Modified, Deleted
+    ChangeType change_type = 2;
+    bool breaking_change = 3;
+  }
+  repeated ChangeLine change_line = 1;
+  int64 change_count = 2;
+  int64 breaking_change_count = 3;
+}
+
+/* This represents the total stats of a resource.
+These stats are computed for all resources. */
+message Stats {
```

Comment:
This is a very generic (too generic) name to use in an "applications" package that contains protos associated with other analysis capabilities. I don't think there's enough here to create a new package, so I'd like to look for message names that include "Stability"

Looking through the messages in this file, I think that "Stats" and "Metrics" could be merged into a single message and if a "filter" field was added, this merged message could replace the last six protos in this file.

Reply:
The "applications" package itself is arbitrary and in some ways temporary (v1alpha1) - I would expect that some applications might grow out of it, but (very subjectively) I think that stability might be one of many factors that could be used to compute an "API Score" and that having these factors in a single package might aid clarity. (I guess I'm saying that I'm not sure that stability belongs in its own package and that even though "applications" probably isn't right either, at least it forces us to think about stability alongside other quantities that might be of interest)